backend/controllers/fieldController.js

```diff
@@ -0,0 +1,114 @@
+const Field = require('../models/FieldSchema');
+const { validationResult } = require('express-validator');
+const asyncHandler = require('express-async-handler');
+const Admin = require("../models/adminSchema");
+
+// @desc Get all fields
+// @route GET /api/fields
+// @access Private -> Admin only
+const getFields = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+  res.send(await Field.find({}));
+});
+
+// @desc Get field by name
+// @route GET /api/fields/:name
+// @access Private -> Admin only
+const getFieldByName = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+
+  constfieldName = req.params.name;
+  const field = await Field.findOne({ [fieldName]: { $exists: true } });
+  if (!field) {
+    return res.status(404).json({ message: "No field found" });
+  }
+
+  res.json({ [fieldName]: field[fieldName] });
+});
+
+// @desc Add item to field
+// @route POST /api/fields
+// @access Private -> Admin only
+const createField = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+
+  const {
+    name, 
+    item,
+  } = req.body;
+
+  let field = new Field({
+    name: name,
+    item: item,
+  });
+
+  try {
+    field = await field.save();
+    res.json(field);
+  } catch (err) {
+    res.status(500).send(err.message);
+    console.log(`error is ${err.message}`);
+  }
+});
+
+// @desc Edit field item
+// @route PUT /api/fields/:name
+// @access Private -> Admin only
+const editField = asyncHandler(async (req, res) => {
+  const admin = await Admin.findById(req.admin.id);
+  if (!admin) {
+    res.status(401);
+    throw new Error("Admin not found");
+  }
+
+  const field = await Field.findOne({ name: req.params.name });
```

Comment:
You won't be able to get the field this way, instead use hasOwnProperty to make sure the name the user passed in is a valid field, then just `Field.findOne()` (you don't need to pass anything in because there is only one instance of 'Field' shared by the entire application)
```
 if (!Object.prototype.hasOwnProperty.call(FieldSchema, name)) {
    res.status(400);
    throw new Error(`Invalid field name ${name}`);
  }
```